pkg/odo/cli/catalog/describe/service.go

```diff
@@ -42,20 +62,88 @@ func NewDescribeServiceOptions() *DescribeServiceOptions {
 
 // Complete completes DescribeServiceOptions after they've been created
 func (o *DescribeServiceOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	isSVCSupported, err := svc.IsCSVSupported()
+	if err != nil {
+		isSVCSupported = false
+	}
+	if strings.Contains(args[0], "/") && isSVCSupported {
+		tmpOptrList := strings.Split(args[0], "/")
+		o.ServiceType = tmpOptrList[0]
+		o.CustomResource = tmpOptrList[1]
+		o.isOperator = true
+	} else {
+		o.serviceName = args[0]
+	}
+
 	o.Context, err = genericclioptions.NewContext(cmd, true)
-	o.serviceName = args[0]
 
 	return
 }
 
 // Validate validates the DescribeServiceOptions based on completed values
 func (o *DescribeServiceOptions) Validate() (err error) {
+
+	if o.isOperator {
+
+		if o.ServiceType == "" || o.CustomResource == "" {
+			return fmt.Errorf("invalid service name, use the format <operator-type>/<crd-name>")
+		}
+		// make sure that CSV of the specified ServiceType exists
+		o.CSV, err = o.KClient.GetClusterServiceVersion(o.ServiceType)
+		if err != nil {
+			// error only occurs when OperatorHub is not installed.
+			// k8s does't have it installed by default but OCP does
+			return err
+		}
+
+		// Get the specific CR that matches "kind"
+		crs := o.KClient.GetCustomResourcesFromCSV(&o.CSV)
+
+		var cr *olm.CRDDescription
+		hasCR := false
+		for _, custRes := range *crs {
+			c := custRes
+			if c.Kind == o.CustomResource {
+				cr = &c
+				hasCR = true
+				break
+			}
+		}
+		if !hasCR {
+			return fmt.Errorf("the %s resource doesn't exist in specified %s operator", o.CustomResource, o.ServiceType)
+		}
+
+		o.CR = cr
+		return nil
+	}
 	o.service, o.plans, err = svc.GetServiceClassAndPlans(o.Client, o.serviceName)
 	return err
+
 }
 
 // Run contains the logic for the command associated with DescribeServiceOptions
 func (o *DescribeServiceOptions) Run() (err error) {
+	if o.isOperator {
+		return o.operatorRun()
```

Comment:
Please create and use interfaces here like it has been done after refactoring for `odo service create` and `odo service delete`. Please refer PR #4465 for more details.

Reply:
I understand the reasoning for that refactor but here dont you think it would be overkill? I this code is quite readable in its current state in my opinion.

Reply:
No, I don't think it would be an overkill because we're actively discussing adding Helm support. Eventually more things might need to be done for Helm integration.

IMO, since we're not on a time crunch on this one, adding interface would be really helpful.

Reply:
done